includes/rules/aria_hidden.php

```diff
@@ -18,11 +18,44 @@ function edac_rule_aria_hidden( $content, $post ) { // phpcs:ignore -- $post is
 	$errors   = array();
 	$elements = $dom->find( '[aria-hidden="true"]' );
 
+	$attributes_that_make_this_valid_hidden_use = array(
+		'class' => 'wp-block-spacer',
+		'role'  => 'presentation',
+	);
+
 	if ( $elements ) {
 		foreach ( $elements as $element ) {
 
-			if ( stristr( $element->getAttribute( 'class' ), 'wp-block-spacer' ) ) {
-				continue;
+			foreach ( $attributes_that_make_this_valid_hidden_use as $attribute => $value ) {
+				if ( stristr( $element->getAttribute( $attribute ), $value ) ) {
+					continue 2;
+				}
+			}
+
+			$parent_node = $element->parent();
+			if ( $parent_node && ( strtolower( $parent_node->tag ) === 'button' || strtolower( $parent_node->tag ) === 'a' ) ) {
```

Comment:
Improve readability

```php
			if ( 
				$parent_node && 
				( 
					strtolower( $parent_node->tag ) === 'button' || 
					strtolower( $parent_node->tag ) === 'a' 
				) 
			) {
```

Reply:
Seeing it laid out like this makes me think that I should reduce the number of lines in the condition. What do you think of this alternative?

```
			$parent_node             = $element->parent();
			$tags_for_further_checks = array(
				'button',
				'a',
			);
			if (
				$parent_node &&
				in_array( strtolower($parent_node->tag), $tags_for_further_checks, true )
			) {
```

Reply:
I'd rather it be a direct comparison than taking on the additional overhead of the `in_array` since the array isn't likely to change. It's a marginal performance improvement but it can add up throughout the code base. I was tempted to ask you to make the `$attributes_that_make_this_valid_hidden_use` a direct comparison as well be I think that one is okay for now.

I know code readability is subjective but I find this more readable anyways.

```php
			if ( 
				$parent_node && 
				( 
					strtolower( $parent_node->tag ) === 'button' || 
					strtolower( $parent_node->tag ) === 'a' 
				) 
			) {
```

Reply:
@SteveJonesDev gotcha, I updated this to align with your suggestion now. LMK if there was any other feedback.